src/PlaneWaveBasis.jl

```diff
@@ -264,8 +269,25 @@ function PlaneWaveBasis(model::Model{T}, Ecut::Real, fft_size::Tuple{Int, Int, I
     # Compute k-point information and spread them across processors
     # Right now we split only the kcoords: both spin channels have to be handled
     # by the same process
-    n_kpt   = length(kcoords_global)
     n_procs = mpi_nprocs(comm_kpts)
+    n_kpt   = length(kcoords_global)
+    n_irreducible_kpoints = n_kpt
+
+    # The code cannot handle MPI ranks without k-points. If there are more prcocesses
+    # than k-points, we duplicate k-points with the highest weight on the empty MPI
+    # ranks (and scale the weight accordingly)
+    if n_procs > n_kpt
+        for i in n_kpt+1:n_procs
+            idx = argmax(kweights_global)
+            kweights_global[idx] *= 0.5
+            push!(kweights_global, kweights_global[idx])
+            push!(kcoords_global, kcoords_global[idx])
+        end
+        @warn("Attempting to parallelize $n_kpt k-points over $n_procs MPI ranks. DFTK does " *
+              "not support processes empty of k-point. Some k-points were duplicated over the " *
+              "extra ranks with scaled weights.")
+    end
+    n_kpt = length(kcoords_global)
 
     if n_procs > n_kpt
```

Comment:
Can't this stuff now be removed ?

Reply:
You mean the warning message? I think it has its use, as over-parallelizing leads to waste of resources.

Reply:
No I mean everything after line 292 in the new line count.

Reply:
Oh yes, for sure!